pages/api/token/temporary.ts

```diff
@@ -0,0 +1,75 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import validate, {
+  BearerTokenPayload,
+  createTemporaryToken,
+  updateTemporaryToken,
+} from "@lib/middleware/bearerToken";
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { formUser } from "@lib/types-database";
+import { QueryResult } from "pg";
+import { logMessage } from "@lib/logger";
+
+const checkRequestPayload = (
+  handler: (
+    req: NextApiRequest,
+    res: NextApiResponse,
+    bearerTokenPayload: BearerTokenPayload,
+    email: string
+  ) => void
+) => {
+  return async (req: NextApiRequest, res: NextApiResponse, options?: unknown): Promise<unknown> => {
+    const requestBody = JSON.parse(req.body);
+    try {
+      if (Object.prototype.hasOwnProperty.call(requestBody, "email")) {
```

Comment:
This is very verbose. Any reason you're not just calling this directly from request body ? Or better yet doing something like `if(requestBody?.email)`